### PR TITLE
Rewrite DashboardPage test using rtl

### LINKED ISSUE #31566: DashboardPage tests: rewrite tests to react-testing-library
Currently, those tests are implemented with Enzymy making them hard to DOM-test. Rewrite them do RTL should make our life easier going forward.

### PR DESCRIPTION
**Pull request by tharun208**
**What this PR does / why we need it**: This PR rewrites the dashboard page tests using react-testing-library

**Which issue(s) this PR fixes**: #31566

Fixes #31566

**Special notes for your reviewer**: I removed some of the tests as it is verifying the internals of the component.

### CODE REVIEW

File: public/app/features/dashboard/containers/DashboardPage.test.tsx [hunk 1]
```diff
@@ -114,138 +159,61 @@ describe('DashboardPage', () => {
...
-      expect(document.title).toBe('My dashboard - Grafana');
+      expect(screen.getByText('My dashboard - Grafana')).toBeInTheDocument();
     });
 
     it('Should render dashboard grid', () => {
       expect(ctx.wrapper).toMatchSnapshot();
     });
   });
 
-  dashboardPageScenario('When user goes into panel edit', (ctx) => {
```
Comment (Clarity-89) [inline public/app/features/dashboard/containers/DashboardPage.test.tsx:129]: These test cases should not be just removed but refactored into tests with RTL. For example, for testing edit mode we could check if the UI elements that visible only in that mode appear on the screen. Same goes for the other removed tests.

Comment (tharun208) [reply public/app/features/dashboard/containers/DashboardPage.test.tsx:129]: sure. will add those 👍🏻 also

Comment (tharun208) [reply public/app/features/dashboard/containers/DashboardPage.test.tsx:129]: @Clarity-89, I am not able to assert for `edit` mode tests.

**Comment by tharun208** [comment]
@Clarity-89, from feedback from this pr https://github.com/grafana/grafana/pull/36226#pullrequestreview-694903041, do we remove the existing snapshots for the dashboard tests also?

**Comment by Clarity-89** [comment]
> @Clarity-89, from feedback from this pr [#36226 (review)](https://github.com/grafana/grafana/pull/36226#pullrequestreview-694903041), do we remove the existing snapshots for the dashboard tests also?

It's always good if snapshots can be removed, but only if they are replaced by other tests (we don't want to decrease test coverage just to remove snapshots). Looking at this component I think rewriting its tests in RTL might be quite a challenge and would probably need tweaking the component itself.

**Comment by tharun208** [comment]
> > @Clarity-89, from feedback from this pr [#36226 (review)](https://github.com/grafana/grafana/pull/36226#pullrequestreview-694903041), do we remove the existing snapshots for the dashboard tests also?
> 
> It's always good if snapshots can be removed, but only if they are replaced by other tests (we don't want to decrease test coverage just to remove snapshots). Looking at this component I think rewriting its tests in RTL might be quite a challenge and would probably need tweaking the component itself.

yes, I am not able to add proper assertions for the `edit mode tests. I will try to complete it.

**Comment by torkelo** [comment]
@tharun208 dont work more on this, picking this up in another PR

**Pull request closed without merging**